### PR TITLE
Add Spring services doc

### PR DESCRIPTION
**Pull request by geofjamg**

### CODE REVIEW

File: docs/micro-services/index.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+```bash
+mvn -Pdocker install
+```
+
+Docker images should be available in your local registry.
+
+```bash
+$ docker images
+powsybl/powsybl-server-discovery                 1.0.0-SNAPSHOT                   b4aa78912a87        49 years ago        163.7 MB
```
Comment (mathbagu) [inline docs/micro-services/index.md]: The `49 years ago` is weird, but I have the same output on my laptop

File: docs/micro-services/index.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+powsybl/powsybl-server-storage                   1.0.0-SNAPSHOT                   5d5dfdc64985        49 years ago        170.9 MB
+powsybl/powsybl-server-storage                   latest                           5d5dfdc64985        49 years ago        170.9 MB
+powsybl/powsybl-server-config                    1.0.0-SNAPSHOT                   33056b76b8a1        49 years ago        145.8 MB
+powsybl/powsybl-server-config                    latest                           33056b76b8a1        49 years ago        145.8 MB
+powsybl/powsybl-server-network                   1.0.0-SNAPSHOT                   c6d5501e3af9        49 years ago        188.1 MB
+powsybl/powsybl-server-network                   latest                           c6d5501e3af9        49 years ago        188.1 MB
+```
+
+Four services have been built:
+ - Config service which is responsible for centralizing configuration of all others services. It is implemented using [Spring Cloud Config](https://spring.io/projects/spring-cloud-config).
```
Comment (mathbagu) [inline docs/micro-services/index.md]: > Config **service is** responsible
> of all **other** services.

File: docs/micro-services/index.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+ - Discovery service which is responsible for service registry. It is implemented using [Netflix Eureka](https://github.com/Netflix/eureka).
```
Comment (mathbagu) [inline docs/micro-services/index.md]: > Discovery **service is** responsible

File: docs/micro-services/index.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+Create `$HOME/.itools/config.yml` and add following configuration to add MapDB storage named 'test':
+
+```yaml
+mapdb-app-file-system:
+        drive-name: test
+        db-file: /storage/db
+        remotely-accessible: true
+```
+
+This file is be shared with server storage container.
```
Comment (mathbagu) [inline docs/micro-services/index.md]: > This file **is shared** with
> with **the** server storage

File: docs/micro-services/index.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+```bash
+docker-compose up
+```
+
+## Getting Swagger documentation
+
+Storage and Network services are documented using [Swagger](https://swagger.io/) and are available once container have been started. 
+ - [Storage service documentation](http://192.168.0.12:8090/swagger-ui.html)
+ - [Network service documentation](http://192.168.0.13:8091/swagger-ui.html)
```
Comment (mathbagu) [inline docs/micro-services/index.md]: These links are not working whereas the containers are up:
```
$> docker container ls
CONTAINER ID        IMAGE                                     COMMAND                  CREATED             STATUS                   PORTS                    NAMES
9e4c54cf3a5b        powsybl/powsybl-server-network:latest     "java -cp /app/resou…"   2 minutes ago       Up 2 minutes (healthy)   0.0.0.0:8091->8091/tcp   powsybl-spring_network-server_1
5976357add29        powsybl/powsybl-server-storage:latest     "java -cp /app/resou…"   2 minutes ago       Up 2 minutes (healthy)   0.0.0.0:8090->8090/tcp   powsybl-spring_storage-server_1
c04e7ca69a92        powsybl/powsybl-server-discovery:latest   "java -cp /app/resou…"   3 minutes ago       Up 3 minutes (healthy)   0.0.0.0:8061->8061/tcp   powsybl-spring_discovery-server_1
3e2c406d6cf6        powsybl/powsybl-server-config:latest      "java -cp /app/resou…"   3 minutes ago       Up 3 minutes (healthy)   0.0.0.0:8088->8088/tcp   powsybl-spring_config-server_
```

The configuration seems OK:
```
$> docker port powsybl-spring_storage-server_1
8090/tcp -> 0.0.0.0:8090

$> docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' powsybl-spring_storage-server_1
192.168.0.12
```

Comment (mathbagu) [reply docs/micro-services/index.md]: The connection works properly if I'm offline (cf. remark on the corresponding [PR](https://github.com/powsybl/powsybl-spring/pull/8))

File: docs/micro-services/index.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+powsybl/powsybl-server-config                    1.0.0-SNAPSHOT                   33056b76b8a1        49 years ago        145.8 MB
+powsybl/powsybl-server-config                    latest                           33056b76b8a1        49 years ago        145.8 MB
+powsybl/powsybl-server-network                   1.0.0-SNAPSHOT                   c6d5501e3af9        49 years ago        188.1 MB
+powsybl/powsybl-server-network                   latest                           c6d5501e3af9        49 years ago        188.1 MB
+```
+
+Four services have been built:
+ - Config service which is responsible for centralizing configuration of all others services. It is implemented using [Spring Cloud Config](https://spring.io/projects/spring-cloud-config).
+ - Discovery service which is responsible for service registry. It is implemented using [Netflix Eureka](https://github.com/Netflix/eureka).
+ - Storage service, is a low level web service responsible for file system like data storage.
```
Comment (mathbagu) [inline docs/micro-services/index.md]: > Storage **service is**
Without the extra comma

Add a hyphen between low and level:
> is a **low-level** web service

File: docs/micro-services/index.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+ - Network service rely on Storage service and allow network data query.
```
Comment (mathbagu) [inline docs/micro-services/index.md]: > Network service **relies on**

File: docs/micro-services/index.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+```yaml
+mapdb-app-file-system:
+        drive-name: test
+        db-file: /storage/db
+        remotely-accessible: true
+```
+
+This file is be shared with server storage container.
+
+Create `$HOME/storage` directory. This directory is also be shared with server storage container and contains MapDB data.
```
Comment (mathbagu) [inline docs/micro-services/index.md]: > is **also shared** with **the** server storage

File: docs/micro-services/index.md (hunk 7 above)
Comment (MioRtia) [reply docs/micro-services/index.md]: Another typo:
> and allow**s** data query

File: docs/micro-services/index.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+ - Config service which is responsible for centralizing configuration of all others services. It is implemented using [Spring Cloud Config](https://spring.io/projects/spring-cloud-config).
+ - Discovery service which is responsible for service registry. It is implemented using [Netflix Eureka](https://github.com/Netflix/eureka).
+ - Storage service, is a low level web service responsible for file system like data storage.
+ - Network service rely on Storage service and allow network data query.
+
+## Services configuration
+
+### AFS configuration
+
+Create `$HOME/.itools/config.yml` and add following configuration to add MapDB storage named 'test':
```
Comment (MioRtia) [inline docs/micro-services/index.md]: > and add **the** following configuration to add **a** MapDB storage name `'test'`

Also add a link to the MapDB filesystem documentation: `[MapDB storage](../afs/afs-mapdb.md)`

File: docs/micro-services/index.md (hunk 1 above)
Comment (jonenst) [reply docs/micro-services/index.md]: the reason for this is explained here https://github.com/GoogleContainerTools/jib/blob/master/docs/faq.md#why-is-my-image-created-48-years-ago

**Pull request merged**